src/tmpfiles/tmpfiles.c

```diff
@@ -1055,8 +1055,12 @@ static int fd_set_perms(
                     fchownat(fd, "",
                              new_uid != st->st_uid ? new_uid : UID_INVALID,
                              new_gid != st->st_gid ? new_gid : GID_INVALID,
-                             AT_EMPTY_PATH) < 0)
-                        return log_error_errno(errno, "fchownat() of %s failed: %m", path);
+                             AT_EMPTY_PATH) < 0) {
+                        if (errno != EROFS)
+                                return log_error_errno(errno, "fchownat() of %s failed: %m", path);
+
+                        log_notice_errno(errno, "%s is read-only, not changing owner to "UID_FMT":"GID_FMT, path, new_uid, new_gid);
```

Comment:
Hmm, the same change shall apply to fchmod_opath calls in this func too?

Reply:
hmm, do we handle gracefully errors like this otherwise? why treat this special? what's the usecase?

Reply:
(maybe binding this to `--graceful` might be ok, but i'd like to understand the usecase for this first)

Reply:
@poettering Arch Linux ships tmpfiles dropins for making /usr/bin/groupmems setgid, which obviously fails on read-only /usr

Reply:
but that raises two questions: why does the file have a wrong group for you, and why is it not important that the thing will suddenly be sgid root rather than sgid some-group?

I mean, that's quite a raise of privilege there, which sounds weird to just ignore?

Reply:
@poettering The raise of privilege is due to mkosi not using `newuidmap` map anymore. I have to make `chown()` a noop with seccomp to make that work which means every setgid binary becomes setgid root. I'm yelling at distributions to get rid of setgid binaries but in the meantime we'll have a few of these lying around

Reply:
hmm, but i think the raise of privs here sounds really dangerous to me, it's not something we should do willy-nilly.

hence, maybe hook this up with --graceful? just document it now does two things instead of just one? 

(or if there's value in keeping this separate from current --graceful, then maybe add a new concept? either a new switch that just downgrades the warning, or maybe even --dont-chown or so, which systematically disables chowning? [because as I understand you, in your case chown()ing can never work])

Reply:
Well it's only valuable for me if it's enabled by default in `systemd-tmpfiles-setup.service`, so neither of those would work for me.

I'll just go ahead and close this. I'll work with the distributions instead to get rid of every setgid I can find.